catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/service/SearchService.java

```diff
@@ -269,7 +261,20 @@ private Query createQuery(Filter filter, Long startIndexLng, Long countLng, Stri
         String sortField = Result.TEMPORAL;
         String sortOrder = DEFAULT_SORT_ORDER;
         Long startIndex = startIndexLng == null ? Long.valueOf(DEFAULT_START_INDEX) : startIndexLng;
-        Long count = countLng == null ? Long.valueOf(DEFAULT_COUNT) : countLng;
+        int count;
+        if (countLng == null) {
```

Comment:
New flow control was needed to account for the possibility a non-null countLng was greater than the maxResultQuery size. I tried to keep the ternary, but it would have resulted in nested operations.